lib/tbot/tbot.go

```diff
@@ -466,3 +480,82 @@ func clientForFacade(
 	c, err := authclient.Connect(ctx, authClientConfig)
 	return c, trace.Wrap(err)
 }
+
+type authPingCache struct {
+	client *auth.Client
+	log    logrus.FieldLogger
+
+	mu          sync.RWMutex
+	fetched     bool
```

Comment:
Do we need `fetched` variable? I think it is `true` when `cachedValue` has a value. If so, it doesn't seem to give us anything.
(same in `proxyPingCache`)

Reply:
Ah true, I went with this to try and avoid sticking a pointer to proto.PingResponse, but that's probably easier and clearer. I'll do that.

Reply:
[24a3fd9](https://github.com/gravitational/teleport/pull/39880/commits/24a3fd922554b661ea1859897dbd06c46461b919)